codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Connection.java

```diff
@@ -61,6 +61,16 @@
         void onStreamClosed(Http2Stream stream);
 
         /**
+         * Notifies the listener that the given stream is now eligible for removal.
+         * The stream may still persist because of dependcies and {@link #onStreamRemoved(Http2Stream)} will
+         * be called after the stream has actually been removed.
+         * <p>
+         * If a {@link RuntimeException} is thrown it will be logged and <strong>not propagated</strong>.
+         * Throwing from this method is not supported and is considered a programming error.
+         */
+        void onStreamEligibleForRemove(Http2Stream stream);
```

Comment:
I don't think we need this ... the closed event seems sufficient for this purpose.